### PR TITLE
feat: add URL parsing support for Resend dashboard links

### PR DESCRIPTION
**Pull request by danilowoz**
## Summary
This PR adds support for accepting Resend dashboard URLs (e.g., `https://resend.com/templates/tmpl_123`) in addition to raw IDs across all resource tools. Users can now copy-paste dashboard URLs directly instead of manually extracting IDs.

## Key Changes
- **New `extractIdFromUrl` utility** (`src/lib/url-parser.ts`): Parses Resend dashboard URLs and extracts resource IDs, with support for broadcasts, templates, and automations. Falls back to treating input as raw ID if not a valid URL.
- **Updated all resource tools** to use the new URL parser:
  - `src/tools/templates.ts`: Updated 7 template operations (get, compose, update, remove, publish, duplicate)
  - `src/tools/broadcasts.ts`: Updated 6 broadcast operations (send, get, remove, compose, update)
  - `src/tools/automations.ts`: Updated 4 automation operations (update, get/list, remove, get runs)
  - `src/tools/editor.ts`: Updated 2 editor operations (get content, create connection)
- **Enhanced documentation**: Updated all tool descriptions and parameter docs to mention URL support with examples
- **Comprehensive test coverage** (`tests/lib/url-parser.test.ts`): 12 test cases covering valid URLs, raw IDs, edge cases (trailing slashes, query parameters, whitespace), and error conditions

## Implementation Details
- The parser validates URLs are from `resend.com` or `www.resend.com` domains only
- Gracefully handles malformed URLs by returning input unchanged
- Supports optional resource type filtering to validate URL matches expected resource
- All tool handlers now accept raw IDs via `rawId`/`rawBroadcastId` parameters and extract the actual ID before API calls
- Backward compatible: raw IDs continue to work exactly as before

https://claude.ai/code/session_015HFP4TWUrdMrkVjB6kWZdA

### CODE REVIEW

File: src/lib/url-parser.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export function extractIdFromUrl(
+  input: string,
+  expectedResource?: 'broadcasts' | 'templates' | 'automations',
+): string {
+  const trimmed = input.trim();
+
+  // Quick check: only attempt URL parsing if it looks like a URL
+  if (!trimmed.startsWith('http://') && !trimmed.startsWith('https://')) {
+    return trimmed;
+  }
```
Comment (felipefreitag) [inline src/lib/url-parser.ts:23]: When extractIdFromUrl receives a recognized resend.com URL but with a mismatched resource type (e.g. a /templates/ URL passed to a broadcast tool), it falls back to returning the full URL string as the API ID. Since a URL is never a valid raw ID, this should throw a descriptive error instead of silently passing garbage to the API. Working on this

**Review by felipefreitag — APPROVED** [review]
